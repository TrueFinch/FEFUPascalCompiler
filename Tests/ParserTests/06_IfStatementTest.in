program a;
var
    i: integer;
begin
    i := 3;
    if (1 + 1 = 2) then
        if (i <> 3) then
            i := 3
        else
            i := 4;
    Write(i);
end.