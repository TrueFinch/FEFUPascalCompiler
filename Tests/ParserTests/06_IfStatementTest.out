                              a                                
                              ┼                                
                          MainBlock                            
       ┌──────────────────────┴───────┐                        
      var                     CompoundStatement                
       ┼         ┌────────────────┬───┴───────────────┐        
 SimpleVarDecl   :=          IfStatement         FunctionCall  
 ┌─────┴┐       ┌┴─┐     ┌────────┴───┐            ┌──┴─┐      
 i  SimpleType  i  3     =       IfStatement     write  i      
       ┌┴              ┌─┴──┐   ┌─────┴─────┐                  
    integer            +    2   <>    :=    :=                 
                      ┌┴─┐     ┌┴─┐  ┌┴─┐  ┌┴─┐                
                      1  1     i  3  i  3  i  4                